README.rst

```diff
@@ -72,6 +51,96 @@ and avoid using an external process::
         f.write(b"Hello")
 
 
+The ``xopen`` function
+----------------------
+
+The ``xopen`` module offers a single function named ``xopen`` with the following
+signature::
+
+  xopen(
+    filename: str | bytes | os.PathLike,
+    mode: Literal["r", "w", "a", "rt", "rb", "wt", "wb", "at", "ab"] = "r",
+    compresslevel: Optional[int] = None,
+    threads: Optional[int] = None,
+    *,
+    encoding: str = "utf-8",
+    errors: Optional[str] = None,
+    newline: Optional[str] = None,
+    format: Optional[str] = None,
+  ) -> IO
+
+The function opens the file using a function suitable for the detected
+file format and returns an open file-like object.
+
+When writing, the file format is chosen based on the file name extension:
+``.gz``, ``.bz2``, ``.xz``, ``.zst``. This can be overriden with ``format``.
+If the extension is not recognized, no compression is used.
+
+When reading and a file name extension is available, the format is detected
+from the extension.
+When reading and no file name extension is available,
+the format is detected from the contents.
+
+Parameters
+~~~~~~~~~~
+
+**filename** (str, bytes, or `os.PathLike <https://docs.python.org/3/library/os.html#os.PathLike>`_):
+Name of the file to open.
+
+If set to ``"-"``, standard output (in mode ``"w"``) or
+standard input (in mode ``"r"``) is returned.
+
+**mode**, **encoding**, **errors**, **newline**:
+These parameters have the same meaning as in Python’s built-in
+`open function <https://docs.python.org/3/library/functions.html#open>`_
+except that the default encoding is always UTF-8 instead of the
+preferred locale encoding.
+``encoding``, ``errors`` and ``newline`` are only used when opening a file in text mode.
+
+**compresslevel**:
+The compression level for writing to gzip, xz and Zstandard files.
+If set to None, a default depending on the format is used:
+gzip: 1, xz: 6, Zstandard: 3.
+
+This parameter is ignored for other compression formats.
+
+**format**:
+Override the autodetection of the input or output format.
+Possible values are: ``"gz"``, ``"xz"``, ``"bz2"``, ``"zst"``.
+
+**threads**:
+If multi-threaded compression or decompression is available,
+this parameter can be used to override the number of threads
+used. It is ignored otherwise.
+
+Set threads to 0 to force opening the file without using a subprocess.
+For some compression levels,
+compressed files are by default read or written
+using a pipe to a subprocess running an external tool such as
+``pbzip2`` or ``xz``.
+With *threads* set to 0, a normal function call is used instead.
```

Comment:
The above is not entirely correct as not always a subprocess is used. Below is my attempt at the paragraph. It is a hard thing to properly explain in a few short sentences so please do not take this as a literal suggestion:

> Sets the number of threads spawned to open a compressed file. May be ignored if the backend does not support threads. 

> By default xopen tries to offload the (de)compression work load to other threads to free up the main Python thread for the application. This can either be done by using a subprocess to an external application or using a library that supports threads.

> Set threads to 0 to force xopen to use only the main Python thread.

Reply:
I changed it to a variant of your suggestion. I agree it’s hard to explain. Maybe an extra section is needed in the documentation somewhere (later). As this paragraph is just the explanation of the parameter, I think it should be relatively short.